File: app/services/teacher.py
import requests
from app.config import Config
URL_BASE = Config.URL_API + '/teacher'


def get_all_teachers(access_token='', page=1, limit=10):
    URL = URL_BASE + f'?page={page}&limit={limit}'

    response = requests.get(
        URL_BASE, headers={'Authorization': 'Bearer ' + access_token})

    if response.status_code != 200:
        return None, response.json()['msg']

    if not response.json():
        return None, "Error en el servidor"

    if response.json()['status'] == "error":
        return None, response.json()['msg']

    return response, None

File: app/static/js/services/teacher.js
import { getTokenCookie } from "../token.js";

const URL_API = "http://localhost:5000/api/";

export const addTeacher = (teacher) => {
  return fetch(`${URL_API}teacher`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: "Bearer " + getTokenCookie(),
    },
    body: JSON.stringify(teacher),
  }).then((response) => {
    if (!response.ok) {
      throw new Error("Network response was not ok");
    }
    return response.json();
  });
};

export const getAllTeacher = (page = 1, limit = 10) => {
  return fetch(`${URL_API}teacher?page=${page}&limit=${limit}`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
      Authorization: "Bearer " + getTokenCookie(),
    },
  }).then((response) => {
    if (!response.ok) {
      throw new Error("Network response was not ok");
    }
    return response.json();
  });
};

export const getTeacher = (id) => {
  return fetch(`${URL_API}teacher/${id}`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
      Authorization: "Bearer " + getTokenCookie(),
    },
  }).then((response) => {
    if (!response.ok) {
      throw new Error("Network response was not ok");
    }
    return response.json();
  });
};
